agents-manage-api/src/routes/agentToolRelations.ts

```diff
@@ -240,15 +240,8 @@ app.openapi(
       });
       return c.json({ data: agentToolRelation }, 201);
     } catch (error) {
-      // Handle foreign key constraint violations
-      if (
-        error instanceof Error &&
-        (error.message.includes('FOREIGN KEY constraint failed') ||
-          error.message.includes('foreign key constraint') ||
-          error.message.includes('SQLITE_CONSTRAINT_FOREIGNKEY') ||
-          (error as any).code === 'SQLITE_CONSTRAINT_FOREIGNKEY' ||
-          (error as any)?.cause?.code === 'SQLITE_CONSTRAINT_FOREIGNKEY')
-      ) {
+      // Handle foreign key constraint violations (PostgreSQL foreign key violation)
+      if ((error as any)?.cause?.code === '23503') {
```

Comment:
Consider adding error handling library to help us map error codes to specific HTTP errors with human readable messages. https://npmtrends.com/pg-error-vs-pg-error-codes-ts-vs-pg-error-enum

Can be done separately.